python/ray/serve/_private/deployment_state.py

```diff
@@ -269,65 +271,54 @@ def actor_handle(self) -> Optional[ActorHandle]:
         return self._actor_handle
 
     @property
-    def version(self) -> Optional[DeploymentVersion]:
+    def version(self) -> DeploymentVersion:
+        """Replica version. This can be incorrect during state recovery.
+
+        If the controller crashes and the deployment state is being
+        recovered, this will temporarily be the deployment-wide target
+        version, which may be inconsistent with the actual version
+        running on the replica actor. If so, the actual version will be
+        updated when the replica transitions from RECOVERING -> RUNNING
+        """
         return self._version
 
     @property
-    def deployment_config(self) -> Optional[DeploymentConfig]:
-        if self._version:
-            return self._version.deployment_config
+    def deployment_config(self) -> DeploymentConfig:
+        """Deployment config. This can return an incorrect config during state recovery.
+
+        If the controller hasn't yet recovered the up-to-date version from the running
+        replica actor, this property will return the target config for the overall
+        deployment.
+        """
+        return self._version.deployment_config
 
     @property
     def max_concurrent_queries(self) -> int:
-        if self.deployment_config:
-            return self.deployment_config.max_concurrent_queries
-        else:
-            # The value in deployment_config won't be respected.
-            # Issue: https://github.com/ray-project/ray/issues/36035
-            logger.warn(
-                "Deployment config is not found, "
-                "using default value for max_concurrent_queries"
-            )
-            return DEFAULT_MAX_CONCURRENT_QUERIES
+        """Maximum number of queries to assign to a replica concurrently.
+        Can return an incorrect value during state recovery.
```

Comment:
no need to duplicate this comment everywhere given you have it in `deployment_config`